src/ngraph/runtime/cpu/pass/cpu_rnn_fusion.cpp

```diff
@@ -62,6 +64,158 @@
+        // Note: ONNX runtime provides W, R and bias in the gate order [IOFC] but
+        // MKLDNN computes LSTM kernel in the [IFCO] order.
+
+        auto get_weights_ifco_gate_order =
+            [&](std::shared_ptr<Node> weights_graph_node) -> std::shared_ptr<Node> {
+            // slices will be in ICFO order
+            std::vector<std::shared_ptr<Node>> gate_slices;
+
+            size_t dim0 = weights_graph_node->get_shape()[0] / 4;
+            size_t dim1 = weights_graph_node->get_shape()[1];
+            for (size_t i = 0; i < 4; i++)
+            {
+                auto slice = std::make_shared<ngraph::op::Slice>(
+                    weights_graph_node, Coordinate{i * dim0, 0}, Coordinate{(i + 1) * dim0, dim1});
+                gate_slices.push_back(slice);
+            }
+
+            auto weights_ifco = std::make_shared<ngraph::op::Concat>(
+                NodeVector{gate_slices[0], gate_slices[2], gate_slices[3], gate_slices[1]}, 0);
+            return weights_ifco;
+        };
+
+        auto get_bias_ifco_gate_order =
+            [&](std::shared_ptr<Node> bias_graph_node) -> std::shared_ptr<Node> {
+
+            size_t hidden_size = lstmcell_op->get_hidden_size();
+            auto Wb_bias = std::make_shared<ngraph::op::Slice>(
+                bias_graph_node, Coordinate{0}, Coordinate{4 * hidden_size});
+            auto Rb_bias = std::make_shared<ngraph::op::Slice>(
+                bias_graph_node, Coordinate{4 * hidden_size}, Coordinate{2 * 4 * hidden_size});
+            auto bias = std::make_shared<op::Add>(Wb_bias, Rb_bias);
+
+            // slices will be in ICFO order
+            std::vector<std::shared_ptr<Node>> gate_slices;
+
+            for (size_t i = 0; i < 4; i++)
+            {
+                auto slice = std::make_shared<ngraph::op::Slice>(
+                    bias, Coordinate{i * hidden_size}, Coordinate{(i + 1) * hidden_size});
+                gate_slices.push_back(slice);
+            }
+
+            auto new_bias = std::make_shared<ngraph::op::Concat>(
+                NodeVector{gate_slices[0], gate_slices[2], gate_slices[3], gate_slices[1]}, 0);
+            return new_bias;
+        };
+
+        auto W_iofc = pattern_map[W];
+        auto R_iofc = pattern_map[R];
+        auto W_ifco = get_weights_ifco_gate_order(W_iofc);
+        auto R_ifco = get_weights_ifco_gate_order(R_iofc);
+        // here onnx bias will be of shape (2 * gates_count * hidden_size) bias of Wb and Rb are concatenated, we will split the bias, add and rearrange in order IFCO
+        auto bias_ifco = get_bias_ifco_gate_order(bias_iofc);
+
+        auto W_reshape = std::make_shared<op::Reshape>(
+            W_ifco, AxisVector{1, 0}, Shape{W_ifco->get_shape()[1], W_ifco->get_shape()[0]});
+        auto R_reshape = std::make_shared<op::Reshape>(
+            R_ifco, AxisVector{1, 0}, Shape{R_ifco->get_shape()[1], R_ifco->get_shape()[0]});
+
+        auto lstm_node = std::make_shared<ngraph::op::Lstm>(
```

Comment:
Please use "#if MKLDNN_VERSION_MAJOR < 1 ... #else ...  #endif" in the fusion to make it support both mkldnn versions.

Reply:
You need #else to create lstm_node for mkldnn-v1.0. lstm has one more input and one more outpu for mkldnn-v1.0.